### PR TITLE
Token Validation API

### PR DESCRIPTION
**Pull request by Kajal-techy**
Hello 
Kindly review my PR.

### CODE REVIEW

File: src/main/java/com/authservice/authservice/config/JwtTokenUtil.java [hunk 1]
```diff
@@ -45,13 +48,75 @@ public String generateToken(User userDetails) {
...
+     *
+     * @param token
+     * @return Boolean
+     */
+    private Boolean isTokenExpired(String token) {
+        final Date expiration = getExpirationDateFromToken(token);
+        return expiration.before(new Date());
+    }
+
+    public Boolean validateToken(String token, UserDetails userDetails) {
```
Comment (AakashIsTheLimit) [inline src/main/java/com/authservice/authservice/config/JwtTokenUtil.java:123]: put java docs on top of this as well. This is the most important function

File: src/main/java/com/authservice/authservice/controller/AuthenticationController.java [hunk 2]
```diff
@@ -44,11 +47,23 @@ public AuthenticationController(AuthenticationManager authenticationManager, Jwt
...
         return ResponseEntity.ok(new JWTResponse(token));
     }
 
+    /**
+     * This function will validate the token and return the userId
+     *
+     * @return
+     */
+    @GetMapping("/validating-token")
+    public ResponseEntity<AuthenticatedResponse> validatingToken(Authentication auth) {
```
Comment (AakashIsTheLimit) [inline src/main/java/com/authservice/authservice/controller/AuthenticationController.java:61]: if you take directly principal it does not work ? are you sure ?

`auth.getPrincipal` you are already doing it does not work with Principal directly ?

File: src/main/java/com/authservice/authservice/exception/Forbidden.java [hunk 3]
```diff
@@ -0,0 +1,8 @@
+package com.authservice.authservice.exception;
+
+public class Forbidden extends RuntimeException {
```
Comment (AakashIsTheLimit) [inline src/main/java/com/authservice/authservice/exception/Forbidden.java]: ForbiddenException

File: src/main/java/com/authservice/authservice/config/JwtTokenUtil.java (hunk 1 above)
Comment (Kajal-techy) [reply src/main/java/com/authservice/authservice/config/JwtTokenUtil.java:123]: Done.

File: src/main/java/com/authservice/authservice/controller/AuthenticationController.java (hunk 2 above)
Comment (Kajal-techy) [reply src/main/java/com/authservice/authservice/controller/AuthenticationController.java:61]: It wouldn't work with Principal.

**Review by AakashIsTheLimit — APPROVED** [review]

**Pull request merged**